build/test/builder/builder_options_test.dart

```diff
@@ -13,11 +13,26 @@ void main() {
           const BuilderOptions(const {'baz': 'different', 'more': 'added'}));
       expect(overridden.config,
           {'foo': 'bar', 'baz': 'different', 'more': 'added'});
+      expect(overridden.isRoot, isFalse);
     });
 
-    test('changes nothing when overriding with empty options', () {
+    test('overrides isRoot', () {
+      var defaults = const BuilderOptions(const {}, isRoot: false);
+      var overridden =
+          defaults.overrideWith(const BuilderOptions(const {}, isRoot: true));
```

Comment:
what do you think of making this const instance a static member of the class?

`defaults.overrideWith(BuilderOptions.forRoot);`

Reply:
sure, why not